darts/models/forecasting/pl_forecasting_module.py

```diff
@@ -100,6 +108,23 @@ def __init__(
             dict() if lr_scheduler_kwargs is None else lr_scheduler_kwargs
         )
 
+        self.metrics = []
+        self.metrics_str = torch_metrics if torch_metrics else []
+        self.metrics_params = metrics_params if metrics_params else []
+
+        if metrics_params:
+            raise_if(
+                len(self.metrics_params) != len(self.metrics_str),
+                "Number of metrics parameters must be equal to number of metrics.",
+                logger,
+            )
+        # create empty dict for each metric
+        if self.metrics_str and metrics_params is None:
+            for _ in self.metrics_str:
+                self.metrics_params.append(dict())
+
+        self._setup_metrics()
```

Comment:
Here `_self_metrics()` has side effects - it relies on the right `self` variables being set (e.g. `self.metrics_str`) and sets `self.metrics`. I would find it cleaner to keep it without side effects (more functional), doing something like
```python
self.metrics = PLForecastingModule._create_metrics(self.metrics_str)
```
with the method static.